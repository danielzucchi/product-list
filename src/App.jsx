import React, { useState, useEffect } from 'react'
import axios from 'axios'
import styled from 'styled-components'
import AppInner from './components/AppInner'

const AppWrapper = styled.div`
  margin: 0;
  width: 100%;
`

const App = () => {
  const [isLoading, setLoading] = useState(false)
  const [fetchError, setError] = useState(null)
  const [products, setProducts] = useState(null)

  useEffect(() => {
    async function fetchProducts() {
      setLoading(true)
      await axios
        .get('https://5c78274f6810ec00148d0ff1.mockapi.io/api/v1/products')
        .then((response) => {
          setLoading(false)
          setProducts(response.data)
          setError(null)
        })
        .catch((error) => {
          setLoading(false)
          setError(`${error.message}. Please refresh page to try again.`)
        })
    }

    fetchProducts()
  }, [setProducts, setError])

  return (
    <AppWrapper>
      <AppInner isLoading={isLoading} products={products} error={fetchError} />
    </AppWrapper>
  )
}

export default App
