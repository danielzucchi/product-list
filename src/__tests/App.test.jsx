import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('App wrapper test', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
