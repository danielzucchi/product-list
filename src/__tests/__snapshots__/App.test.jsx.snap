// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App wrapper test renders without crashing 1`] = `
<styled.div>
  <AppInner
    error={null}
    isLoading={false}
    products={null}
  />
</styled.div>
`;
